pkg/dogstatsd/server.go

```diff
@@ -120,6 +121,23 @@ type metricsCountBuckets struct {
 	closeChan  chan struct{}
 }
 
+func getTelemetryEndpoint(endpoint string, originDetection bool, packetsChannel chan listeners.Packets, sharedPacketPool *listeners.PacketPool) (listeners.StatsdListener, error) {
+	if strings.HasPrefix(endpoint, "unix://") {
+		socketPath := endpoint[len("unix://"):]
+		log.Debugf("enabling telemetry on UDS socket %s", socketPath)
+		return listeners.NewUDSListener(socketPath, originDetection, packetsChannel, sharedPacketPool)
+	} else if listeners.IsNamedPipeEndpoint(endpoint) {
+		log.Debugf("enabling telemetry on named pipe %s", endpoint)
+		return listeners.NewNamedPipeListener(endpoint, packetsChannel, sharedPacketPool)
+	}
+	portNumber, err := strconv.Atoi(endpoint)
+	if err != nil && portNumber > 0 {
```

Comment:
`if err != nil && portNumber > 0 {`

The `if err != nil` is suspicious.

Reply:
Why ? We're catching the case of a bad configuration where the telemetry is set to `abc` for example.

Reply:
You are testing that there is an error and if that's the case, you're using the read `portNumber` within `endpoint` to start a listener.
You want to test `err == nil` instead.

Reply:
IMO reversing the logic would even be better.

```go
	portNumber, err := strconv.Atoi(endpoint)
	if err != nil || portNumber == 0 {
		return nil, fmt.Errorf("invalid telemetry udp port '%s': %s", endpoint, err.Error())
	}
	log.Debugf("enabling telemetry on UDP port %d", portNumber)
	return listeners.NewUDPListener(portNumber, packetsChannel, sharedPacketPool)
```